src/cdk/lib/constructs/assets.ts

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+import { RemovalPolicy, Stack } from 'aws-cdk-lib';
+import { Bucket } from 'aws-cdk-lib/aws-s3';
+import { Construct } from 'constructs';
+import { BucketDeployment, Source } from 'aws-cdk-lib/aws-s3-deployment';
+import { NagSuppressions } from 'cdk-nag';
+
+/**
+ * Properties for configuring Assets construct
+ * @interface AssetsProperties
+ */
+export interface AssetsProperties {
+    /**
+     * Array of local file paths to deploy to the S3 bucket as seed data
+     * Files will be deployed to the 'petimages' prefix in the bucket
+     * @optional
+     */
+    seedPaths?: string[];
+}
+
+/**
+ * AWS CDK Construct that creates S3 bucket for pet adoption assets with optional seed data deployment
+ * @class Assets
+ * @extends Construct
+ */
+export class WorkshopAssets extends Construct {
+    /**
+     * The S3 bucket for storing pet adoption assets
+     * @public
+     */
+    public bucket: Bucket;
+
+    /**
+     * Creates a new Assets construct with S3 bucket and optional seed data deployment
+     * If seedPaths are provided, deploys local assets to the bucket under 'petimages' prefix
+     * @param scope - The parent construct
+     * @param id - The construct ID
+     * @param properties - Configuration properties for the construct
+     */
+    constructor(scope: Construct, id: string, properties?: AssetsProperties) {
+        super(scope, id);
+
+        this.bucket = new Bucket(this, 'petadoptionBucket', {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The bucket should have server access logging enabled to provide detailed records for the requests that are made to the bucket.